lib/md051.js

```diff
@@ -0,0 +1,36 @@
+// @ts-check
+
+"use strict";
+
+const { pathToFileURL } = require("url");
+const fs = require("fs");
+
+const { filterTokens, addError } = require("../helpers");
+
+module.exports = {
+  "names": [ "MD051", "no-dead-relative-links" ],
+  "description": "No dead relative links",
+  "tags": [ "links" ],
+  "function": function MD051(params, onError) {
+    filterTokens(params, "inline", (token) => {
+      token.children.forEach((child) => {
+        const { lineNumber, type, attrs } = child;
+        if (type === "link_open") {
+          attrs.forEach((attr) => {
+            if (attr[0] === "href") {
+              const href = attr[1];
+              const url = new URL(href, pathToFileURL(params.name));
+              url.hash = "";
```

Comment:
Currently, "we don't care" about the hash, we only check if the file exists but we could do beyond that and also check if the file is a markdown file and if we have a `href` with a hash (e.g: `./README.md#about`), we could check that we actually have a heading with this hash (e.g: `about`).
	
This would fix this issue: https://github.com/DavidAnson/markdownlint/issues/253
I will try to fix that issue in this PR (not yet implemented).

### Potential Implementation

Taking the example of this `href`:  `./README.md#about`
We could read the file content `./README.md`, parse all the headings in an array following the algorithm as specified in this comment: https://github.com/DavidAnson/markdownlint/issues/253#issuecomment-651369616 and then check if this array contains the hash `about`?
Is this would be a viable solution? Simply ask for guidance and open the discussion. :smile: